packages/webpack-cli/lib/groups/ConfigGroup.js

```diff
@@ -58,162 +64,163 @@ const getConfigInfoFromFileName = (filename) => {
+        } else {
+            newOptionsObject['options'] = namedOptions;
+        }
+    } else {
+        if (Array.isArray(configOptions) && !configOptions.length) {
+            newOptionsObject['options'] = {};
+            return newOptionsObject;
         }
+        newOptionsObject['options'] = configOptions;
     }
 
-    async resolveConfigMerging() {
-        const { merge } = this.args;
-        if (merge) {
-            // Get the current configuration options
-            const { options: configOptions } = this.opts;
-
-            // we can only merge when there are multiple configurations
-            // either by passing multiple configs by flags or passing a
-            // single config exporting an array
-            if (!Array.isArray(configOptions)) {
-                throw new ConfigError('Atleast two configurations are required for merge.', 'MergeError');
-            }
-
-            // We return a single config object which is passed to the compiler
-            const mergedOptions = configOptions.reduce((currentConfig, mergedConfig) => webpackMerge(currentConfig, mergedConfig), {});
-            this.opts['options'] = mergedOptions;
+    if (configOptions && configPath.includes('.webpack')) {
+        const currentPath = configPath;
+        const parentContext = dirname(currentPath).split(sep).slice(0, -1).join(sep);
+        if (Array.isArray(configOptions)) {
+            configOptions.forEach((config) => {
+                config.context = config.context || parentContext;
+            });
+        } else {
+            configOptions.context = configOptions.context || parentContext;
         }
+        newOptionsObject['options'] = configOptions;
     }
+    return newOptionsObject;
+};
 
-    async run() {
-        await this.resolveConfigFiles();
-        await this.resolveConfigMerging();
-        return this.opts;
+const resolveConfigMerging = async (args) => {
+    const { merge } = args;
+    if (merge) {
+        // Get the current configuration options
+        const { options: configOptions } = opts;
+
+        // we can only merge when there are multiple configurations
+        // either by passing multiple configs by flags or passing a
+        // single config exporting an array
+        if (!Array.isArray(configOptions)) {
+            throw new ConfigError('Atleast two configurations are required for merge.', 'MergeError');
+        }
+
+        // We return a single config object which is passed to the compiler
+        const mergedOptions = configOptions.reduce((currentConfig, mergedConfig) => webpackMerge(currentConfig, mergedConfig), {});
+        opts['options'] = mergedOptions;
     }
-}
+};
+
+const handleConfigResolution = async (args) => {
+    const parsedArgs = arrayToObject(args);
+    await resolveConfigFiles(parsedArgs);
+    await resolveConfigMerging(parsedArgs);
+    return opts;
+};
 
-module.exports = ConfigGroup;
+module.exports = handleConfigResolution;
```

Comment:
I think we need move this from `ConfigGroup.js` file, to be honestly we don't need `groups` files :smile: Do you want to do it in this PR?

Reply:
Yep lets do it here

Reply:
Or maybe let's migrate all groups first then we can move them all from `groups` to `argResolvers`?

Reply:
Maybe we put each function in own file, anyway let's refactor firstly and when we can think about places

Reply:
yep once all groups are migrated, we can organize it neatly